evaluation.py

```diff
@@ -237,7 +237,97 @@ def predict_keypoints(self,generator, location):
                 image_ids.append(metadata['src_set_image_id'])
                 j+=1
         self.write_to_json_file(list_of_predictions, location)
-        return image_ids
+        return image_ids, list_of_predictions
+
+    def pck_eval(self, generator, location, annotation_file):
+        _, list_of_predictions = self.predict_keypoints(generator, location)
+        f = open(annotation_file,)
+        data = json.load(f)
+
+        dist_list = []
+        correct_keypoints = {
+            "nose": 0,
+            "left_eye": 0,
+            "right_eye": 0,
+            "left_ear": 0,
+            "right_ear": 0,
+            "left_shoulder": 0,
+            "right_shoulder": 0,
+            "left_elbow": 0,
+            "right_elbow": 0,
+            "left_wrist": 0,
+            "right_wrist": 0,
+            "left_hip": 0,
+            "right_hip": 0,
+            "left_knee": 0,
+            "right_knee": 0,
+            "left_ankle": 0,
+            "right_ankle": 0
+        }
+
+        for prediction in list_of_predictions:
+            prediction_image_id = prediction['image_id']
+            prediction_keypoints = prediction['keypoints']
+            for i, ann in enumerate(data['annotations']):
+                if data['annotations'][i]['image_id'] == prediction_image_id:
+                    annotation_keypoints = data['annotations'][i]['keypoints']
+                    prediction_keypoints = np.array(prediction_keypoints)
+                    annotation_keypoints = np.array(annotation_keypoints)
+
+                    # Joint at 11 is left hip, Joint at 12 is right hip
+                    left_hip_point = np.array(annotation_keypoints[33], annotation_keypoints[34])
+                    right_hip_point = np.array(annotation_keypoints[36], annotation_keypoints[37])
+                    # PCK@0.2 == Distance between predicted and true joint < 0.2 * torso diameter
```

Comment:
I used this threshold for now because it was listed here https://github.com/cbsudux/Human-Pose-Estimation-101#percentage-of-correct-key-points---pck not sure if we want to use another threshold or what they mean by 50% of the head bone link haha

Reply:
Sounds good to me! Maybe throw the threshold in `constants`?

Reply:
So it adjusts the torso length for each image, I could add just the 0.2 to `constants`, what do you think?

Reply:
Sounds good, or even at the top of this file too. Just so it's a bit easier to find in the future